testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/AbstractServletsAdapterTest.java

```diff
@@ -229,4 +247,55 @@ protected void setAdapterServletTimeOffset(int timeOffset, String servletUri) {
         log.info(pageSource);
     }
 
+    /**
+     * Get adapter configuration from file
+     *
+     * @param configJSON URL of configuration file
+     * @return adapter configuration
+     */
+    protected static AdapterConfig getConfig(URL configJSON) {
+        if (configJSON == null) return null;
+
+        try {
+            return getSystemPropertiesObjectMapper().readValue(configJSON, AdapterConfig.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * Update configuration file for servlet
+     *
+     * @param configFile    URL of configuration file
+     * @param adapterConfig consumer with required changes in configuration file
+     * @return updated configuration file
+     */
+    protected static File updateJsonConfigFile(URL configFile, Consumer<AdapterConfig> adapterConfig) {
+        if (configFile == null) return null;
+
+        File file = new File(configFile.getFile());
+        AdapterConfig config = getConfig(configFile);
+
+        if (config != null && adapterConfig != null) {
+            adapterConfig.accept(config);
+
+            try {
+                getSystemPropertiesObjectMapper().writeValue(file, config);
+            } catch (IOException e) {
+                throw new RuntimeException("Cannot write adapter configuration", e);
+            }
+        }
+        return file;
+    }
+
+    /**
+     * Custom Object mapper
+     */
+    private static ObjectMapper getSystemPropertiesObjectMapper() {
+        ObjectMapper mapper = new ObjectMapper(new SystemPropertiesJsonParserFactory());
```

Comment:
`ObjectMapper` should be a static constant rather than recreated each time.

Reply:
You're right.